code/game/objects/items/devices/megaphone.dm

```diff
@@ -13,7 +13,7 @@
 	var/list/insultmsg = list("FUCK EVERYONE!", "I'M A TATER!", "ALL SECURITY TO SHOOT ME ON SIGHT!", "I HAVE A BOMB!", "CAPTAIN IS A COMDOM!", "FOR THE SYNDICATE!")
 
 /obj/item/megaphone/attack_self(mob/living/user as mob)
-	if(user.client && (user.client.prefs.muted & MUTE_IC))
+	if(check_mute(user.ckey, MUTE_IC))
```

Comment:
As far as I know you cant even run an attack_self without being connected so /shrug